tests/testthat/test-rules.R

```diff
@@ -116,3 +116,115 @@ test_that("as.list and rule are reversible when .to_NA is NULL", {
   test_rule <- rule(a = c("a", "b"), b = c("c", "d"), .drop = FALSE, .na_last = TRUE, .to_NA = NULL)
   expect_identical(do.call(rule, as.list(test_rule)), test_rule)
 })
+
+# combine_rules ----
+
+test_that("combine_rules works as expected", {
+  r1 <- rule(a = "1", b = "2", .to_NA = "x", .drop = TRUE, .na_last = FALSE)
+  r2 <- rule(a = "3", c = "4", .to_NA = "y", .drop = FALSE)
+
+  res <- combine_rules(r1, r2)
+  expect_s3_class(res, "rule")
+  expect_identical(res, rule(a = "3", c = "4", b = "2", .to_NA = "y", .drop = FALSE, .na_last = TRUE))
```

Comment:
why the order is `a`, `c`, and `b`?

Reply:
should we, natually have `a` and `b` first (from rule 1), and then other elements in rule 2?

Reply:
I decided to place first the mapping from the rule that has the priority (here rule2). We could have an option to apply an alphabetic sorting.

Reply:
Do you think we should go the other way around @clarkliming ?